internal/gatewayapi/listener.go

```diff
@@ -112,6 +112,12 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 						EscapedSlashesAction: ir.UnescapeAndRedirect,
 					},
 				}
+				if len(listener.frontendValidationCACerts) > 0 {
```

Comment:
We also need to add frontendValidationCaCerts to TLS.CACertificate for TLS listeners.